test/memory/benmcd/oomCheck.chpl

```diff
@@ -0,0 +1,7 @@
+use MemDiagnostics;
+
+const size = here.physicalMemory()*2;
+
+const dom = {0..#size};
+
+var myArr = dom.tryBuildArray(int);
```

Comment:
I think there should be a test that catches the error and responds to it in some way.

I think it would also be good if there was a test that looked for memory leaks in a situation like this, to see if we leak any of the Block-related classes somehow.

Reply:
Looks like there are memory leaks! Will work to get those fixed, good catch.